### PR TITLE
dev/user-interface#49  - SearchKit: improve tabbed interface

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
This implements a bunch of the suggestions in https://lab.civicrm.org/dev/user-interface/-/issues/49 to reorganize the SK tabs.

Before
----------------------------------------
- Monolithic "Compose Search" tab.
- Top row is the title, entity, save button & forms
- Results table indented by the tab nav.

image

After
----------------------------------------
- Compose Search split into multiple tabs for different part of the search.
- Top row is title, save button, view button & forms; entity moved into the "search for" tab.
- Results table placed below tab nav for full-width.
- Select fields tab allows adding a field and reordering fields (same as column headers, but less cramped).

image

### CODE REVIEW

**Comment by vingle** [comment]
That looks a big improvement. 

Is it possible to make the tabs accessible? Civi's JQuery tabs on the contact dashboard are accessible (ie keyboard controllable, tab to select, arrow to move between) tho I don't think the Bootstrap3 tabs are.

It doesn't look like there's unique ID's for each tab/panel - which is what ARIA wants (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role) trickier. I couldn't see from Angular's [a11y docs](https://docs.angularjs.org/guide/accessibility) if they simplify the process of this. But at a least making each tab a `<button>` should make them keyboard target-able, and giving them `role=tab` rather than `role=presentation` (and the panels with `role=tabpanel`) might at least indicate to screen readers what this part of the page is supposed to be doing.

**Comment by aydun** [comment]
I like it.  Design isn't my thing so I'll leave it open for now for others to weigh in but it seems simple and straightforward.

One slight downside of this layout is that you can't see all the parts of the query on the screen at the same time, so eg screenshots on SE may need multiple screenshots in place of the current one.  But that's not sufficient reason to hold back this change.

I wonder whether a few lines of boxing around tab headers and the content panel would make the tabbed structure more obvious.

**Comment by colemanw** [comment]
> @vingle Is it possible to make the tabs accessible?

This PR takes a step in that direction by wrapping the tab `<ul>` inside a `<nav>` element, which is [what the BS3 docs suggested](https://getbootstrap.com/docs/3.4/components/#nav) for accessibility. But even before doing that, I'm able to navigate them with the keyboard. Pressing tab goes to the next, er, tab, and pressing enter opens it. Am I missing something?

**Comment by colemanw** [comment]
> @aydun I wonder whether a few lines of boxing around tab headers and the content panel would make the tabbed structure more obvious.

I agree. But I'm not sure where my role as developer ends and the role of a themer begins. I tried to do everything with vanilla BS3. But if it makes sense we could add more styling?

**Comment by vingle** [comment]
> But even before doing that, I'm able to navigate them with the keyboard. Pressing tab goes to the next, er, tab, and pressing enter opens it. Am I missing something?

@colemanw I'm not getting that on Firefox. Have just checked Safari and the same there. On both: keyboard tabbing skips the tabs and doesn't open it. 

NB - I see from the link that its BS3 recommending `role='presentation'`. But I think `role='tabs'` and `'tablist'` weren't available in 2013; as well as MDN, the W3C ARIA guidelines [currently recommends](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/examples/tabs-manual/) using tabs, tablist and tabpanel - none of which should have any impact on what Bootstrap3 is doing here.

**Comment by vingle** [comment]
RE styling - what about adding Bootstrap Panel classes to the flex wrapper? ie `<div class="crm-flex-4 panel panel-default panel-body">` to get the below:

image

**Comment by colemanw** [comment]
@vingle I've committed your suggestions, and with a touch more css was able to achieve a flowy look between the tabs and the panel.

File: ext/search_kit/ang/crmSearchAdmin/crmSearchAdmin.html [hunk 1]
```diff
@@ -90,15 +90,15 @@ <h1 crm-page-title>{{ $ctrl.entityTitle + ': ' + $ctrl.savedSearch.label }}</h1>
...
 
-    <div class="crm-flex-box">
+    <div class="crm-flex-box crm-search-admin-outer">
       <nav class="crm-search-admin-main-tabs">
         <ul class="nav nav-pills nav-stacked" role="tablist" aria-orientation="vertical" ng-include="'~/crmSearchAdmin/tabs.html'"></ul>
       </nav>
-      <div class="crm-flex-4">
-        <div ng-if="!$ctrl.selectedDisplay()">
+      <div class="crm-flex-4 panel panel-default">
+        <div ng-if="!$ctrl.selectedDisplay()" class="panel-body">
```
[the remark this answers is not in the document]
Comment (colemanw) [reply ext/search_kit/ang/crmSearchAdmin/crmSearchAdmin.html]: Thanks @vingle - fixed.

**Comment by vingle** [comment]
@colemanw Looks much improved.

**Comment by colemanw** [comment]
Thanks @vingle, I just pushed a few more tweaks (namely shifting the tabs down a few pixels so the rounded corner of the tab pane stays round) & updated the description.

**Comment by vingle** [comment]
@colemanw I'm still getting the line overlap on the merge demo (It's probably different with different themes)…

image

**Comment by colemanw** [comment]
@vingle ok I just pushed an update. I can't say I understand where that misalignment is coming from in your testing, but what I just pushed should be more forgiving.

**Comment by vingle** [comment]
> @vingle ok I just pushed an update. I can't say I understand where that misalignment is coming from in your testing, but what I just pushed should be more forgiving.

It's because all the other `<li>`s have their left margin set to 0. Only the first `<li>` gets the 0.5em yellow margin below:

image

With latest commit (NB - this is around 1400px width viewport):

image

With `margin-left: 0;`:

image

**Comment by colemanw** [comment]
Ok @vingle and I resolved this over chat & seems to be good now.
I think this PR is now merge-ready.

**Comment by aydun** [comment]
I like those updates.  A good step forward for the UI!

**Comment by UshaMakoa** [comment]
Nice! https://github.com/civicrm/civicrm-core/pull/28740 works fine!
Thanks [@colemanw](https://github.com/colemanw)

**Pull request merged**